pkg/signup/service/signup_service.go

```diff
@@ -142,14 +143,14 @@ func extractEmailHost(email string) string {
 	}
 
 	return encoded
-}*/
+}
 
 // Signup reactivates the deactivated UserSignup resource or creates a new one with the specified username and userID
 // if doesn't exist yet.
 func (s *ServiceImpl) Signup(ctx *gin.Context) (*v1alpha1.UserSignup, error) {
-	userID := ctx.GetString(context.SubKey)
+	encodedUserID := EncodeUserID(ctx.GetString(context.SubKey))
 	// Retrieve UserSignup resource from the host cluster
-	userSignup, err := s.CRTClient().V1Alpha1().UserSignups().Get(userID)
+	userSignup, err := s.CRTClient().V1Alpha1().UserSignups().Get(encodedUserID)
```

Comment:
It means we can't just change the encoding in the future without a migration step. Let's add a comment to the encoding function to warn about any possible change in the function that would change the user id encoding.

Reply:
I've added a warning, I'll commit the change after e2e tests have run